pkg/scheduler/factory/factory.go

```diff
@@ -208,36 +215,22 @@ func NewConfigFactory(
 				switch t := obj.(type) {
 				case *v1.Pod:
 					return unassignedNonTerminatedPod(t)
+				case cache.DeletedFinalStateUnknown:
+					if pod, ok := t.Obj.(*v1.Pod); ok {
+						return unassignedNonTerminatedPod(pod)
+					} else {
+						runtime.HandleError(fmt.Errorf("unable to convert object %T to *v1.Pod in %T", obj, c))
+						return false
+					}
 				default:
 					runtime.HandleError(fmt.Errorf("unable to handle object in %T: %T", c, obj))
 					return false
 				}
 			},
 			Handler: cache.ResourceEventHandlerFuncs{
-				AddFunc: func(obj interface{}) {
```

Comment:
I moved these to their own separate functions and added better object type handling for "DeleteFunc".